### PR TITLE
Add null annotations to SubscriptionProvider

### PR DESCRIPTION
**Pull request by wborn**
Without these annotations the build will fail when https://github.com/openhab/openhab-core/pull/1412 gets merged.

### CODE REVIEW

File: bundles/org.openhab.ui.habot/src/main/java/org/openhab/ui/habot/notification/internal/webpush/Subscription.java [hunk 1]
```diff
@@ -24,21 +24,21 @@
...
         this.endpoint = endpoint;
         this.keys = keys;
     }
 
     public class Keys {
-        public String p256dh;
+        public @NonNullByDefault({}) String p256dh;
 
-        public String auth;
+        public @NonNullByDefault({}) String auth;
```
Comment (wborn) [inline bundles/org.openhab.ui.habot/src/main/java/org/openhab/ui/habot/notification/internal/webpush/Subscription.java:41]: Do you know if the default constructor is unnecessary @ghys?
If so, we can remove it and make these fields final without this annotation.

Comment (ghys) [reply bundles/org.openhab.ui.habot/src/main/java/org/openhab/ui/habot/notification/internal/webpush/Subscription.java:41]: Hi @wborn, yes it appears to be unnecessary. It actually comes from https://github.com/web-push-libs/webpush-java/blob/master/src/main/java/nl/martijndwars/webpush/Subscription.java, I don't remember the exact reason why I didn't simply use the library, I think I needed some changes.

Comment (wborn) [reply bundles/org.openhab.ui.habot/src/main/java/org/openhab/ui/habot/notification/internal/webpush/Subscription.java]: Gone it is! :+1:

**Comment by wborn** [comment]
You might also want to merge this @openhab/webui-maintainers because builds will likely fail since https://github.com/openhab/openhab-core/pull/1412 got merged.

**Comment by ghys** [comment]
Thank you @wborn!

**Pull request merged**